src/Build/BuildCheck/Infrastructure/BuildCheckConnectorLogger.cs

```diff
@@ -79,22 +91,47 @@ private void EventSource_AnyEventRaised(object sender, BuildEventArgs e)
         }
     }
 
-    private readonly Dictionary<string, TimeSpan> _stats = new Dictionary<string, TimeSpan>();
+    private readonly Dictionary<string, TimeSpan> _statsInfra = new Dictionary<string, TimeSpan>();
+    private readonly Dictionary<string, TimeSpan> _statsAnalyzers = new Dictionary<string, TimeSpan>();
 
     private void EventSource_BuildFinished(object sender, BuildFinishedEventArgs e)
     {
-        _stats.Merge(buildCheckManager.CreateTracingStats(), (span1, span2) => span1 + span2);
-        string msg = string.Join(Environment.NewLine, _stats.Select(a => a.Key + ": " + a.Value));
-
 
         BuildEventContext buildEventContext = e.BuildEventContext ?? new BuildEventContext(
             BuildEventContext.InvalidNodeId, BuildEventContext.InvalidTargetId,
             BuildEventContext.InvalidProjectContextId, BuildEventContext.InvalidTaskId);
 
         LoggingContext loggingContext = loggingContextFactory.CreateLoggingContext(buildEventContext).ToLoggingContext();
 
-        // TODO: tracing: https://github.com/dotnet/msbuild/issues/9629
-        loggingContext.LogCommentFromText(MessageImportance.High, msg);
+        if (_areStatsEnabled)
+        {
+            _statsAnalyzers.Merge(buildCheckManager.CreateAnalyzerTracingStats(), (span1, span2) => span1 + span2);
+            _statsInfra.Merge(buildCheckManager.CreateBuildCheckInfraTracingStats(), (span1, span2) => span1 + span2);
+
+            LogAnalyzerStats(loggingContext);
+        }
+    }
+    
+    private void LogAnalyzerStats(LoggingContext loggingContext)
+    {
+        loggingContext.LogCommentFromText(MessageImportance.High, $"BuildCheck run times{Environment.NewLine}");
+        string infraData = buildStatsTable("Infrastructure run times", _statsInfra);
+        loggingContext.LogCommentFromText(MessageImportance.High, infraData);
+
+        string analyzerData = buildStatsTable("Analyzer run times", _statsAnalyzers);
+        loggingContext.LogCommentFromText(MessageImportance.High, analyzerData);
+    }
+
+    private string buildStatsTable(string title, Dictionary<string, TimeSpan> rowData)
+    {
+        string headerSeparator = $"=============";
```

Comment:
nit: ```suggestion
        string headerSeparator = "=============";
```
Do we need this?

Reply:
I was thinking of it as a visual barrier. As we don't have connection to the binlog yet this is the only place the analyzer stats will be displayed. This list can also have the potential to increase by a lot so I think that for now this header is a good separator. Definitely not forever though, since the final form of the instrumentation module will be shown on bonlog / binlog viewer.

Reply:
Thank you for the clarification!